curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/shopping-list.english.md

```diff
@@ -55,22 +55,20 @@ var myList = [];
 ```js
 var count = 0;
 var isArray = false;
-var hasString = false;
-var hasNumber = false;
+var hasString = true;
+var hasNumber = true;
 (function(list){
   if(Array.isArray(myList)) {
     isArray = true;
     if(myList.length > 0) {
-      hasString = true;
-      hasNumber = true;
-      myList.forEach(function(elem) {
-        if(typeof elem[0] !== 'string') {
+      for (var elem of myList) {
+        if(!elem || typeof elem[0] !== 'string') {
```

Comment:
```suggestion
        if(!elem || !elem[0] || typeof elem[0] !== 'string') {
```

Reply:
This will ensure empty strings don't pass the test.

Reply:
@Manish-Giri Good catch!